index.js

```diff
@@ -63,12 +51,12 @@ async function makeSitemap(opts = {}) {
   }
   // Creates a sitemap object given the input configuration with URLs
   const sitemap = sm.createSitemap(options)
-  // Generates XML with a callback function
-  sitemap.toXML(error => {
-    if (error) {
-      throw error
-    }
-  })
+  // Generates XML
+  try {
+    await util.promisify(sitemap.toXML.bind(sitemap))()
+  } catch (error) {
+    failBuild('Could not generate XML sitemap', { error })
```

Comment:
What does this do exactly?

Reply:
This calls `sitemap.toXML()` and waits for it to complete.

If `sitemap.toXML()` errors, it reports it with `utils.build.failBuild()` in order to display it in build logs.

Reply:
Gotcha. It exits the build correct? We don't want sites deploying without sitemaps

Reply:
Yes it fails the build.

Reply:
ok cool. sounds good